include/moqt/publish_track_handler.h

```diff
@@ -0,0 +1,267 @@
+
+        /**
+         * @brief set/update the default TTL expiry for published objects
+         */
+        void SetDefaultTTL(uint32_t ttl) { def_ttl_ = ttl; }
+
+        /**
+         * @brief Get the publish status
+         *
+         * @return Status of publish
+         */
+        Status GetStatus() { return publish_status_; }
+
+        // --------------------------------------------------------------------------
+        // Methods that normally do not need to be overridden
+        // --------------------------------------------------------------------------
+
+        /**
+         * @brief Publish object to the announced track
+         *
+         * @details Publish an object to announced track that was previously announced.
+         *   This will have an error if the track wasn't announced yet. Status will
+         *   indicate if there are no subscribers. In this case, the object will
+         *   not be sent.
+         *
+         *   **Restrictions:**
+         *   - This method cannot be called twice with the same object header group and object IDs.
+         *   - In TrackMode::kStreamPerGroup, ObjectHeaders::group_id **MUST** be different than the previous
+         *     when calling this method when the previous has not been completed yet using PublishContinuationData().
+         *     If group id is not different, the PublishStatus::kPreviousObjectNotCompleteMustStartNewGroup will be returned
+         *     and the object will not be sent. If new group ID is provided, then the previous object will terminate
+         *     with stream closure, resulting in the previous being truncated.
+         *   - In TrackMode::kStreamPerTrack, this method **CANNOT** be called until the previous object has been
+         *     completed using PublishContinuationData(). Calling this method before completing the previous
+         *     object remaining data will result in PublishStatus::kObjectDataIncomplete. No data would be sent and the
+         *     stream would remain unchanged. It is expected that the caller would send the remaining continuation data.
+         *
+         * @note If data is less than ObjectHeaders::payload_length, then PublishObject()
+         *   should be called to send the remaining data.
+         *
+         * @param object_headers        Object headers, must include group and object Ids
+         * @param data                  Payload data for the object, must be <= object_headers.payload_length
+         *
+         * @returns Publish status of the publish
+         *    * PublishStatus::kObjectContinuationDataNeeded if the object payload data is not completed but it was sent,
+         *    * PublishStatus::kObjectDataComplete if the object data was sent and the data is complete,
+         *    * other PublishStatus
+         */
+        PublishStatus PublishObject(const ObjectHeaders& object_headers, BytesSpan data);
```

Comment:
not sure of the reason to use Span here. Isn't the entire object gets moved/copied over within the function ? Span is helpful when you want to read into and not for cases when you take over the ownership. May be I am missing how are the parameters used.
Also rename data to object_payload.

Reply:
No.  It's not copied/moved because that introduced unnecessary copies of the same object for each subscriber and  client. The construct of using span vs vector work transparently. You can pass a vector or you can pass `Bytes` another span `BytesSpan`. 

The implementation is not under review here, but the implementation handles this to ensure that when this method returns, the use of the object data is done.  We pass this down a few times to get it to the transport for transmission, that is why we had unnecessary copies, including the initial for each publish object fan out for all subscribers.  It is in the final transmission queue where we do a single copy for transmission. That isn't easily avoidable.

Reply:
I was not advocating for copy but rather move .. Move semantics are not expensive. If we decide to not do that then the API should document saying the caller shouldn't free the resource until the function returns.